Fails with Quarkus `2.12.0.CR1` due to `ProblemProcessor#warnOnMissingSmallryeMetricsDependency`

**Describe the bug**
```java
2022-08-17 15:46:26,104 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus:
java.lang.RuntimeException: java.lang.IllegalArgumentException:
Build step 'com.tietoevry.quarkus.resteasy.problem.deployment.ProblemProcessor#warnOnMissingSmallryeMetricsDependency'
does not produce any build item and thus will never get executed.

Either change the return type of the method to a build item type, add a parameter of type
BuildProducer<[some build item type]>/Consumer<[some build item type]>,
or annotate the method with @Produces. Use @Produce(EmptyBuildItem.class) if you want to always execute this step.

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:330)                                                                                                                                     
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)                                                                                                                 
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:87)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:448)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:60)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:148)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:103)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)

Caused by: java.lang.IllegalArgumentException: Build step 'com.tietoevry.quarkus.resteasy.problem.deployment.ProblemProcessor#warnOnMissingSmallryeMetricsDependency' does not produce any build item and thus will never get executed.

Either change the return type of the method to a build item type, add a parameter of type BuildProducer<[some build item type]>/Consumer<[some build item type]>, or annotate the method with @Produces. Use @Produce(EmptyBuildItem.class) if you want to always execute this step.
```

**To Reproduce**
Steps to reproduce the behavior:
- Sample application or code (github repo / gist / inline)
- Quarkus version, quarkus-resteasy-problem version

**Expected behavior**
A clear and concise description of what you expected to happen.

--- a/.github/workflows/latest-stable-compatibility-tests.yaml
+++ b/.github/workflows/latest-stable-compatibility-tests.yaml
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         json-provider: [ "jackson-classic", "jsonb-classic", "jackson-reactive", "jsonb-reactive" ]
-        quarkus-version: [ "2.0", "2.1", "2.2", "2.3", "2.4", "2.5", "2.6", "2.7", "2.8", "2.9", "2.10"]
+        quarkus-version: [ "2.0", "2.1", "2.2", "2.3", "2.4", "2.5", "2.6", "2.7", "2.8", "2.9", "2.10", "2.11"]
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/maven-regression.yaml
+++ b/.github/workflows/maven-regression.yaml
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        quarkus-version: [ "2.0", "2.1", "2.2", "2.3", "2.4", "2.5", "2.6", "2.7", "2.8", "2.9", "2.10"]
+        quarkus-version: [ "2.0", "2.1", "2.2", "2.3", "2.4", "2.5", "2.6", "2.7", "2.8", "2.9", "2.10", "2.11"]
         json-provider: [ "jsonb-classic", "jackson-classic" ]
 
     steps:

--- a/README.md
+++ b/README.md
@@ -47,7 +47,7 @@ so-called "HTTP APIs" are usually not.
 ### Quarkus 2.X / Java 11+
 Make sure JDK 11 is in your PATH, the run:
 ```shell
-mvn io.quarkus:quarkus-maven-plugin:2.11.2.Final:create \
+mvn io.quarkus:quarkus-maven-plugin:2.12.0.Final:create \
     -DprojectGroupId=problem \
     -DprojectArtifactId=quarkus-resteasy-problem-playground \
     -DclassName="problem.HelloResource" \

--- a/deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java
+++ b/deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java
@@ -153,14 +153,6 @@ public class ProblemProcessor {
         }
     }
 
-    @BuildStep(onlyIfNot = QuarkusSmallryeMetricsDetector.class)
-    void warnOnMissingSmallryeMetricsDependency(ProblemBuildConfig config) {
-        if (config.metricsEnabled) {
-            logger().warn("quarkus.resteasy.problem.metrics.enabled is set to true, but quarkus-smallrye-metrics not "
-                    + "found in the classpath");
-        }
-    }
-
     protected Logger logger() {
         return LoggerFactory.getLogger(FEATURE_NAME);
     }

--- a/integration-test/pom.xml
+++ b/integration-test/pom.xml
@@ -228,6 +228,13 @@
                 <quarkus.version>2.10.3.Final</quarkus.version>
             </properties>
         </profile>
+        
+        <profile>
+            <id>quarkus-2.11</id>
+            <properties>
+                <quarkus.version>2.11.2.Final</quarkus.version>
+            </properties>
+        </profile>
 
         <profile>
             <id>native</id>

--- a/integration-test/pom.xml
+++ b/integration-test/pom.xml
@@ -232,7 +232,7 @@
         <profile>
             <id>quarkus-2.11</id>
             <properties>
-                <quarkus.version>2.11.2.Final</quarkus.version>
+                <quarkus.version>2.11.3.Final</quarkus.version>
             </properties>
         </profile>
 

--- a/pom.xml
+++ b/pom.xml
@@ -19,7 +19,7 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
         <!-- runtime/test dependencies -->
-        <quarkus.version>2.12.0.CR1</quarkus.version>
+        <quarkus.version>2.12.0.Final</quarkus.version>
         <zalando-problem.version>0.27.1</zalando-problem.version>
         <assertj.version>3.23.1</assertj.version>
         <jmh.version>1.35</jmh.version>

--- a/pom.xml
+++ b/pom.xml
@@ -19,7 +19,7 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
         <!-- runtime/test dependencies -->
-        <quarkus.version>2.11.2.Final</quarkus.version>
+        <quarkus.version>2.12.0.CR1</quarkus.version>
         <zalando-problem.version>0.27.1</zalando-problem.version>
         <assertj.version>3.23.1</assertj.version>
         <jmh.version>1.35</jmh.version>

